test/e2e/common/node/image_credential_provider.go

```diff
@@ -48,11 +51,22 @@ var _ = SIGDescribe("ImageCredentialProvider", feature.KubeletCredentialProvider
 	ginkgo.It("should be able to create pod with image credentials fetched from external credential provider ", func(ctx context.Context) {
 		privateimage := imageutils.GetConfig(imageutils.AgnhostPrivate)
```

Comment:
@SergeyKanzhelev @haircommander Are there any other private images that we can use for tests with the test `gcp-credential-provider`? If there is another image, I can configure that in `matchImages` for the symlink-ed provider that uses SA tokens. That way we can confirm image pull worked because that plugin was the one that did the validation and fetched the token (using the default flow which is un-changed).

I have confirmed the provider with the SA token validation works by having a commit that only has `gcp-credential-provider-with-sa` configured:
- https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128372/pull-kubernetes-node-kubelet-credential-provider/1854258712361832448

Reply:
I am not seeing any other private images so I don't believe there's another. I'm not really familiar with how private images are setup in CI but naively I can imagine a new one could be added. This could be a follow up IMO

Reply:
sounds good! I had summarized that here https://github.com/kubernetes/kubernetes/pull/128372#discussion_r1832149131. I will do this in a follow-up.